### PR TITLE
Changes related to folder prefix.

### PR DESCRIPTION
**Pull request by kwwaikar**

### CODE REVIEW

**Review by dbarrundiag — COMMENTED** [review]
This is great...

Can we just add a bit more details on both of the logs when listing the keys under the prefix?

Something like:
```
logging.info(f"Adding key to manifest: {file['Key']}")
```

and not sure if we need to log the `page` but if it's helpful then we can also add more details:
```
logging.info(f"Finding keys in prefix={prefix} and page={page}")
```

**Comment by dbarrundiag** [comment]
Also i feel like we should just delete all the "product finalize" code that is not relevant anymore

File: source/StartPublishingWorkflowFunction/app.py [hunk 1]
```diff
@@ -38,7 +38,30 @@ def lambda_handler(event, context):
...
+        for entry in intial_asset_list:
+            asset_bucket=entry['Bucket']
+            prefix=entry['Key']
+            if (prefix.endswith('/')):
+                paginator = s3.get_paginator('list_objects_v2')
+                response_iterator = paginator.paginate(Bucket=asset_bucket,Prefix=prefix,PaginationConfig={'PageSize': 1000})
+                for page in response_iterator:
+                    logging.info(page)
+                    if 'Contents' not in page:
+                        return {
```
Comment (dbarrundiag) [inline source/StartPublishingWorkflowFunction/app.py:53]: We should raise an exception here and let the final except-block catch the exception:

```
    except Exception as error:
        logging.error(f'lambda_handler error: {error}')
        logging.error(f'lambda_handler trace: {traceback.format_exc()}')
        result = {
            'Error': f'{error=}'
        }
        return json.dumps(result)
```

Comment (kwwaikar) [reply source/StartPublishingWorkflowFunction/app.py:53]: fixed in commit

**Comment by kwwaikar** [comment]
> Also i feel like we should just delete all the "product finalize" code that is not relevant anymore

Lets leave it for now Incase providers who have not migrated the products want to use it.

File: source/StartPublishingWorkflowFunction/app.py (hunk 1 above)
Comment (dbarrundiag) [reply source/StartPublishingWorkflowFunction/app.py:53]: not seeing the commit, maybe we are missing a push?

**Comment by kwwaikar** [comment]
Hi Daniel, i have updated the code. Please see the latest commit.

**Review by dbarrundiag — APPROVED** [review]

**Pull request merged**